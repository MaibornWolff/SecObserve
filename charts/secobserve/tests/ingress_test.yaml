suite: test ingress
templates:
  - ingress.yaml
release:
  name: test-release
  namespace: test-namespace
tests:
  - it: check manifest with custom host
    set:
      ingress:
        enabled: true
        ingressClassName: alb
        hostname: test.example.com
    asserts:
    - matchSnapshot: {}

