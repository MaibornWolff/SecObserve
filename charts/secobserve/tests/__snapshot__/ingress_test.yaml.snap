check manifest with custom host:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        alb.ingress.kubernetes.io/healthcheck-path: /
        alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS": 443}]'
        alb.ingress.kubernetes.io/ssl-policy: ELBSecurityPolicy-TLS13-1-2-FIPS-2023-04
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1.30.1
        com.secobserve.tenant: secobserve
        helm.sh/chart: secobserve-1.0.0
      name: secobserve-ing
      namespace: test-namespace
    spec:
      ingressClassName: alb
      rules:
        - host: test.example.com
          http:
            paths:
              - backend:
                  service:
                    name: secobserve-svc
                    port:
                      name: backend-http
                path: /api/
                pathType: Prefix
              - backend:
                  service:
                    name: secobserve-svc
                    port:
                      name: backend-http
                path: /django-static/
                pathType: Prefix
              - backend:
                  service:
                    name: secobserve-svc
                    port:
                      name: frontend-http
                path: /
                pathType: Prefix
